pkg/server/server_sql.go

```diff
@@ -1459,6 +1459,18 @@ func (s *SQLServer) preStart(
 	if err := s.settingsWatcher.Start(ctx); err != nil {
 		return errors.Wrap(err, "initializing settings")
 	}
+
+	// Prevent a secondary tenant's server from starting if its binary version is too low
+	// for the current tenant cluster version.
+	if !s.execCfg.Codec.ForSystemTenant() {
```

Comment:
It would be better to structure this code to not need this codec check here. 
The check should be the same for the system tenant.

Reply:
Actually based on how you structured the code, i'd say just drop the check and the remainder of the code is correct.

Reply:
Done